tlscommon.c

```diff
@@ -152,6 +426,35 @@ static int verify_cb(int ok, X509_STORE_CTX *ctx) {
         X509_STORE_CTX_set_error(ctx, err);
     }
 
+    ssl = X509_STORE_CTX_get_ex_data(ctx, SSL_get_ex_data_X509_STORE_CTX_idx());
+    tlsconf = (struct tls *)SSL_get_ex_data(ssl, RADSEC_TLS_EX_INDEX_TLSCONF);
+    if(tlsconf != NULL && tlsconf->ocspcheck) {
+        if(tlsconf->ocsp_check_depth && (tlsconf->ocsp_check_depth >= depth)) {
+            debug(DBG_DBG, "verify_cb: skipping OCSP Chck for certificate %i, ocsp check depth reached");
+        }
+        else {
+            issuer_cert = X509_STORE_CTX_get0_current_issuer(ctx);
+            if (issuer_cert == NULL) {
+                // Issuer is unknown. We can't verify OCSP with unknown issuer.
+                if (ok != 0) {
+                    debug(DBG_WARN, "verify error: Unknown issuer, skipping OCSP Check");
+                    ok = 0;
+                }
+            } else if (issuer_cert == err_cert) {
+                // if the cert is selfsigned we don't need ocsp
+            } else {
+              store = (X509_STORE *)SSL_get_ex_data(ssl, RADSEC_TLS_EX_INDEX_STORE);
```

Comment:
I vaguely remember that I had a specific reason for doing it this way.
It may be that the reason was "I cant get the store from the CTX" because I did not find this function, but I'm not entirely sure.